### ISSUE TITLE
Domino MixtureSlicer results disagree with sklearn.mixture.GaussianMixtureModel

### ISSUE THREAD

**Issue by Robdei** [HazyResearch/domino#16]
I might be missing something, but it seems like there might be something off in the Mixture model implementation in Domino. In the tutorial, we take the first 512 images from the imaginette validation set and calculate CLIP embeddings + Resnet predictions then create slices.
```
df = mk.datasets.get("imagenette")
df = df[df["split"] == "valid"][:512]
df = embed(df, input_col="img",encoder="clip", modality="image",device=DEVICE)
(* resnet prediction and target code here *)
domino = DominoSlicer(y_log_likelihood_weight=0,y_hat_log_likelihood_weight=0,n_mixture_components=25,n_slices=25,
    init_params='kmeans',n_pca_components=None) #set weights to zero so that the mixture model only fits to clip embeddings + disable PCA
domino.fit(data=df, embeddings="clip(img)", targets="target", pred_probs="prob")
df["domino_slices"] = domino.predict_proba(data=df, embeddings="clip(img)", targets="target", pred_probs="prob")
print(np.array(df["domino_slices"]).argmax(axis=1))
```
which yields these cluster predictions:
`array([22, 13,  2,  2, 20, 20,  1, 20,  5, 20, 20, 10, 23, 23, 13,  2,  1, 14,  2, 20, 16, 18, 13, 23, 15, 18, 18, 22,  5, 16,  3, 12,  5, 23,...])`

However, if we use the same embeddings and settings for `sklearn.mixture.GaussianMixture`:
```
from sklearn import mixture
X = np.array(df['clip(img)'])
gmm = mixture.GaussianMixture(n_components=25, covariance_type='full', init_params='kmeans')
gmm.fit(X)
print(gmm.predict(X))
```

Which yields these cluster predictions:
`array([ 0,  8, 20, 19, 19, 12, 10, 19,  0, 12, 14, 21,  7,  7,  8, 11,  0, 20,  2, 14, 13, 12,  8, 14, 19, 12, 12, 13,  0, 13, 17,  2, 13,  7...])`

These predictions are significantly different. Is it possible there's an issue in the MixtureSlicer?
